### ISSUE TITLE
Area lights much darker in final render than in viewport render

### ISSUE THREAD

**Issue by Theverat** [LuxCoreRender/BlendLuxCore#321]
LuxCore version: 243366f3e1b9bed96b6b3d36a4d53e3caced3216
BlendLuxCore version: 0787bfec77cb67d551a86b6cefd1112decab44cb

In final render, area lights are much darker than they should be. In viewport they are correct.
Other light types work fine.
Test scene: [viewport_final_difference.zip](https://github.com/LuxCoreRender/BlendLuxCore/files/3825693/viewport_final_difference.zip)

I am filing this as an addon bug for now, although I am not sure if it is not a LuxCore bug.
The problem does not appear if I force area lights to be instanced in final render, by replacing this line:
https://github.com/LuxCoreRender/BlendLuxCore/blob/0787bfec77cb67d551a86b6cefd1112decab44cb/export/light.py#L399
with `if True:`.
But so far, I could not replicate this outside of Blender with a scene written by FILESAVER (those always work correctly).

Area light (final on the left, viewport on the right):

![area](https://user-images.githubusercontent.com/1584761/68499740-3bb12a00-025a-11ea-91e7-6754a98b6d60.PNG)

Point light:

![point](https://user-images.githubusercontent.com/1584761/68499747-3fdd4780-025a-11ea-9ad5-0e09fb655a35.PNG)

**Comment by Theverat** [comment]
@Dade916 If you have any ideas about this, I would be very glad, because I'm at my wits end :/

**Comment by Dade916** [comment]
I will try to check tomorrow what is going on.

The area light comparison is particularly puzzling, there is really a huge difference.

**Comment by Dade916** [comment]
Note: with FILESAVER, the mesh is saved to file while, inside Blender, the translation is done on-the-fly in ram. It is a quite different procedure, there may be some mis-alignment between the two.
I could have fixed/changed something inside LuxCore and forget to update Blender dedicated code. It looks like something related to transformations.

**Comment by Theverat** [comment]
Yes, that could be, or maybe I screwed something up while porting the mesh conversion code to Blender 2.80.

**Comment by Dade916** [comment]
The problem is in the computation of the triangle area (used to estimate the total emission), it is 4 when you do the final rendering and 0.0144 in all other cases (including with LuxCoreUI). I'm tracking why.

**Comment by Dade916** [comment]
Ok, I should have fixed the problem: for the final render, you were defining the mesh and than applying a transformation to vertices however the second step wasn't updating the pre-compute total mesh area (required if the transformation has a scale factor).

**Comment by Theverat** [comment]
Awesome, thanks a lot!
